src/extension.ts

```diff
@@ -196,9 +200,49 @@ function registerDockerClients(): void {
         if (e.affectsConfiguration('docker.composeCommand')) {
             composeClient.reconfigure();
         }
+
+        if (e.affectsConfiguration('docker.contexts.showInStatusBar')) {
+            dockerContextStatusBarItem.dispose();
+            // Don't wait
+            void registerContextStatusBarItems(ext.context);
+        }
     });
 }
 
+async function registerContextStatusBarItems({ subscriptions }: vscode.ExtensionContext) {
```

Comment:
In retrospect I think we _should_ put this into its own file, along with the configuration change event listener. `utils` folder is fine I think.

Reply:
good suggestion, will do!